lib/matplotlib/scale.py

```diff
@@ -279,7 +329,8 @@ class LogScale(ScaleBase):
     """
     name = 'log'
 
-    def __init__(self, axis, *, base=10, subs=None, nonpositive="clip"):
+    @_make_axis_parameter_optional
+    def __init__(self, axis=None, *, base=10, subs=None, nonpositive="clip"):
```

Comment:
Is there a reason this was changed from `axis` to `axis=None`, while _some_ of the others weren't?

Reply:
We normally cannot re-order parameters, as that would break callers, though the decorator may avoid that problem. Looking back at previous discussion, I see that @timhoffm kept these parameter/argument passing for derived classes, so I suppose we can't just delete it entirely and will have to leave them around for now.